core/scripts/ocr2vrf/util.go

```diff
@@ -248,6 +253,37 @@ func getRandomness(e helpers.Environment, coordinatorAddress string, requestID *
 	helpers.ConfirmTXMined(context.Background(), e.Ec, tx, e.ChainID)
 }
 
+func requestRandomnessFromConsumer(e helpers.Environment, consumerAddress string, numWords uint16, subID uint64, confDelay *big.Int) *big.Int {
+	consumer := newVRFBeaconCoordinatorConsumer(common.HexToAddress(consumerAddress), e.Ec)
+
+	tx, err := consumer.TestRequestRandomness(e.Owner, numWords, subID, confDelay)
+	helpers.PanicErr(err)
+	receipt := helpers.ConfirmTXMined(context.Background(), e.Ec, tx, e.ChainID)
+
+	periodBlocks, err := consumer.IBeaconPeriodBlocks(nil)
+	helpers.PanicErr(err)
+
+	blockNumber := receipt.BlockNumber
+	periodOffset := new(big.Int).Mod(blockNumber, periodBlocks)
+	nextBeaconOutputHeight := new(big.Int).Sub(new(big.Int).Add(blockNumber, periodBlocks), periodOffset)
+
+	fmt.Println("nextBeaconOutputHeight: ", nextBeaconOutputHeight)
+
+	requestID, err := consumer.SRequestsIDs(nil, nextBeaconOutputHeight, confDelay)
```

Comment:
So this returns the most recent requestID for that beacon height/confDelay combo, right?

Reply:
yeah that's right. seems like the consumer contract implementation overrides the requestID for the same height/confDelay